### DISCUSSION TITLE
How to subscribe to events fired by `useWeb3ModalEvents()` hook ?

### DISCUSSION THREAD

**Discussion by rameshdon1** [WalletConnect/walletconnect-monorepo#3991] [Developer Support]
In the example folder of web3modal, we have very basic example of useWeb3ModalEvents hook. 
![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/139542777/a3a2e434-a6bd-4f0b-811d-17bbd8b3acfb)

And the output looks like below on UI

![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/139542777/5083f1e8-6cb9-44fd-afe1-83e03dafd27a)

But if i have to listen for events and when an event is fired, lets say "CONNECT_SUCCESS", then how can I listen to that and do action based on it?

Should i have to match the event string like `if(events.data.event === "CONNECT_SUCCESS") { }` on useEffect or i can do something like `event.on("CONNECT_SUCCESS",()=>{ })`. Am I missing something ?

I am using NextJs and Wagmi, Here is my configuration that is wrapping components on layout.tsx,

 `
      'use client';

    import { configureChains, createConfig, WagmiConfig } from 'wagmi';
    import { polygonMumbai } from 'wagmi/chains';
    import { CoinbaseWalletConnector } from 'wagmi/connectors/coinbaseWallet';
    import { jsonRpcProvider } from 'wagmi/providers/jsonRpc';

    import { getChainId, getNetworkUrl } from '@/utils/magicNetwork';
    import { DedicatedWalletConnector } from '@magiclabs/wagmi-connector';
    import { WalletConnectConnector } from '@wagmi/core/connectors/walletConnect';
    import { createWeb3Modal, EIP6963Connector } from '@web3modal/wagmi/react';

    const WalletConnect = ({ children }: { children: React.ReactNode }) => {
    const { chains, publicClient, webSocketPublicClient } = configureChains(
    [polygonMumbai],
    [
      jsonRpcProvider({
        rpc: () => ({
          http: `${process.env.NEXT_PUBLIC_MUMBAI_RPC_URL}`,
        }),
      }),
     ]
     );

     const projectId = '32xxxxxxxxxxxx';

     const metadata = {
      name: 'Test',
      description: 'Test',
      url: 'https://test.com',
      icons: ['https://avatars.githubusercontent.com/u/37784886'],
     };

     const wagmiConfig = createConfig({
     autoConnect: true,
     connectors: [
      new WalletConnectConnector({
        chains,
        options: { projectId, showQrModal: false, metadata },
      }),
      new CoinbaseWalletConnector({
        chains,
        options: { appName: metadata.name },
      }),
      new EIP6963Connector({ chains }),
     ],
     publicClient,
     webSocketPublicClient,
     });

     const modal =   createWeb3Modal({
     wagmiConfig: wagmiConfig,
     projectId,
     chains,
     featuredWalletIds: [''],
     });

  

     return <WagmiConfig config={wagmiConfig}>{children}</WagmiConfig>;
     };

     export default WalletConnect;
`

 Please help me on this. Thanks

**Comment by glitch-txs** [comment]
Hi, you can use Wagmi hooks instead. For connection state you can check [useAccount hook](https://wagmi.sh/react/hooks/useAccount) for example
